crates/gosub_html5/src/tokenizer.rs

```diff
@@ -337,25 +323,28 @@ impl<'stream> Tokenizer<'stream> {
                         }
                         Ch(ch @ 'A'..='Z') => self.add_to_token_name(to_lowercase!(ch)),
                         Ch(CHAR_NUL) => {
-                            self.parse_error(ParserError::UnexpectedNullCharacter);
+                            self.parse_error(ParserError::UnexpectedNullCharacter, loc);
                             self.add_to_token_name(CHAR_REPLACEMENT);
                         }
-                        Eof => {
-                            self.parse_error(ParserError::EofInTag);
+                        StreamEnd => {
+                            self.parse_error(ParserError::EofInTag, loc);
                             self.state = State::Data;
                         }
                         _ => self.add_to_token_name(c.into()),
                     }
                 }
                 State::RCDATALessThanSign => {
                     let c = self.read_char();
-                    if let Ch('/') = c {
-                        self.temporary_buffer.clear();
-                        self.state = State::RCDATAEndTagOpen;
-                    } else {
-                        self.consume('<');
-                        self.chars.unread();
-                        self.state = State::RCDATA;
+                    match c {
+                        Ch('/') => {
+                            self.temporary_buffer.clear();
+                            self.state = State::RCDATAEndTagOpen;
+                        }
+                        _ => {
+                            self.consume('<');
+                            self.stream_prev();
+                            self.state = State::RCDATA;
+                        }
```

Comment:
I guess, this should be a `if let` match pattern again, in my clippy PR, this actually will be an warning.

Reply:
fixed